### DISCUSSION TITLE
Linechart - no dataset displayed

### DISCUSSION THREAD

**Discussion by howudodat** [pepstock-org/Charba#64] [Q&A] [answered]
Just getting started using this in a GWT/dominokit project.  I will be retrieving fft signal data from a db and then plotting it. The below code is just a simple test.  I get the chart and the red legend, the axis is scaled down to about -60 and the title changes, but no line is drawn.

```
         protected void initUI() {
		Charba.enable();
		chart = new LineChart();
		
		chart.getOptions().setResponsive(true);
		chart.getOptions().getLegend().setPosition(Position.TOP);
		chart.getOptions().getTitle().setDisplay(true);
		chart.getOptions().getTitle().setText("Charba line chart");
		
		this.getContentElement().element().appendChild(chart.getChartElement().as());
		testChart();
	}
	
	protected void testChart() {
		LinkedList<Double> l = new LinkedList<>();
		l.add(-45.1);
		l.add(-42.1);
		l.add(-45.1);
		l.add(-39.1);
		
		LineDataset dataset = new LineDataset();
		dataset.setData(l);
		dataset.setBackgroundColor(HtmlColor.RED);
		chart.getData().setDatasets(dataset);
		chart.getOptions().getTitle().setText("Test");
		chart.update();
	}
```

UPDATE: the problem appears to be that we dont have any labels set.  We will probably only set about 5 labels, but we will have 200+ points of data.  This is a frequency plot and there will not be enough room to display all the labels

UPDATE2: This is getting uglier.  If I set 3 labels, only 3 points are plotted.  What I really would like is something like: labels["0", "50", "100"] and they be spaced at the start, middle and end, but all of the points get plotted.

**Comment by howudodat** [comment, marked as answer]
ok, quick code to help anyone that might follow
```
		CartesianCategoryAxis axis1 = new CartesianCategoryAxis(chart);
		axis1.setDisplay(true);
		axis1.getTitle().setDisplay(false);
		axis1.getTicks().setCallback(new CategoryTickCallback() {
			@Override
			public String onCallback(Axis axis, String value, int index, List<String> values) {
				return "";
			}
		});
		chart.getOptions().getScales().setAxes(axis1);
```

**Comment by stockiNail** [comment]
@howudodat the data to load in a chart should have always 2 dimensions, for cartesian axes as you are using.

- `value`: the value of dataset that you want to show (line chart has cartesian linear axis as default). The values can be usually set in a dataset by `setData` method .
- `index`: the value to show on index axis, which are represented by ticks. The index can be usually set by `chart.getData().setLabel`  or in a dataset by `setLabels` method .

Furthermore you can `DataPoint` (with x and y properties), avoiding to set labels where for each value you have a specific index.

In the example, the `value` is set as list of numbers but the `index` is missing, therefore the line is not drawn.

The code that you published, all ticks (related to the index) are not showed .

To have a clear picture about your expected result, can you publish here how the chart should look like?
In this case I can prepare a sample and published here, if you want.

I'm glad you are using Charba on DominoKit. Let me know, if you want, how it's working because I haven't tested it yet.
